python/shared.py

```diff
@@ -0,0 +1,129 @@
+import json
+from dataclasses import dataclass, field
+from typing import Optional, Iterable, Any
+
+from dataclasses_json import config
+
+TASK_QUEUE = "CustomerLoyaltyTaskQueue"
+CUSTOMER_WORKFLOW_ID_FORMAT = "customer-{}"
+EVENT_HISTORY_THRESHOLD = 10000
+
+# Signal and query names
+SIGNAL_CANCEL_ACCOUNT = "cancelAccount"
+SIGNAL_ADD_POINTS = "addLoyaltyPoints"
+SIGNAL_INVITE_GUEST = "inviteGuest"
+SIGNAL_ENSURE_MINIMUM_STATUS = "ensureMinimumStatus"
+QUERY_GET_STATUS = "getStatus"
+QUERY_GET_GUESTS = "getGuests"
+
+
+@dataclass
+class Customer:
+    customerId: str = field(metadata=config(field_name="customerId"))
+    name: Optional[str] = ""
+    points: Optional[int] = 0
+    guests: Optional[set[str]] = field(default_factory=set)
```

Comment:
```suggestion
    guests: set[str] = field(default_factory=set)
```

No need for optional here if it's never gonna be set to None. Also I'd recommend a list (even though it's O(n) for dupe check, at least order is predictable)